tests/Aspire.RabbitMQ.Client.Tests/AspireRabbitMQLoggingTests.cs

```diff
@@ -0,0 +1,183 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Logging;
+using Testcontainers.RabbitMq;
+using Xunit;
+
+namespace Aspire.RabbitMQ.Client.Tests;
+
+public class AspireRabbitMQLoggingTests
+{
+    /// <summary>
+    /// Tests that the RabbitMQ client logs are forwarded to the M.E.Logging correctly in an end-to-end scenario.
+    ///
+    /// The easiest way to ensure a log is written is to start the RabbitMQ container, establish the connection,
+    /// and then stop the container. This will cause the RabbitMQ client to log an error message.
+    /// </summary>
+    [LocalOnlyFact]
+    public async Task EndToEndLoggingTest()
+    {
+        await using var rabbitMqContainer = new RabbitMqBuilder().Build();
+        await rabbitMqContainer.StartAsync();
+
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+        builder.Configuration.AddInMemoryCollection([
+            new KeyValuePair<string, string?>("ConnectionStrings:messaging", rabbitMqContainer.GetConnectionString())
+        ]);
+
+        builder.AddRabbitMQ("messaging");
+
+        var tsc = new TaskCompletionSource();
+        var logger = new TestLogger();
+        logger.LoggedMessage = () =>
+        {
+            // wait for at least 2 logs to be written
+            if (logger.Logs.Count >= 2)
+            {
+                tsc.SetResult();
+            }
+        };
+
+        builder.Services.AddSingleton<ILoggerProvider>(sp => new LoggerProvider(logger));
+
+        using var host = builder.Build();
+        using var connection = host.Services.GetRequiredService<IConnection>();
+
+        await rabbitMqContainer.StopAsync();
+
+        await tsc.Task;
```

Comment:
Should there be a timeout or are we certain some log will be written eventually?

Reply:
Added a 1 minute timeout.